libcxx/include/__numeric/saturation_arithmetic.h

```diff
@@ -0,0 +1,119 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___NUMERIC_SATURATION_ARITHMETIC_H
+#define _LIBCPP___NUMERIC_SATURATION_ARITHMETIC_H
+
+#include <__concepts/arithmetic.h>
+#include <__config>
+#include <__type_traits/decay.h>
+#include <__utility/cmp.h>
+#include <limits>
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#  pragma GCC system_header
+#endif
+
+_LIBCPP_BEGIN_NAMESPACE_STD
+
+#if _LIBCPP_STD_VER >= 26
+
+template <typename _Tp>
+concept __libcpp_standard_integer = __libcpp_unsigned_integer<decay_t<_Tp>> || __libcpp_signed_integer<decay_t<_Tp>>;
```

Comment:
I think we should use `remove_cv_t` here. The current standard wording says
> _Constraints_: `T` is a signed or unsigned integer type ([basic.fundamental]).

which clearly rejects reference types.

(Also the name seems misleading as it's intended to accept extended integer types.)

----
BTW, after looking into `<__utility/cmp.h>`, it seems to me that the newly added `__libcpp_standard_integer` should be essentially same as `__is_safe_integral_cmp`.

https://github.com/llvm/llvm-project/blob/7dd4d28e4196fad83ed78ea342d65e7eaec4a6f1/libcxx/include/__utility/cmp.h#L35-L51

Perhaps we can only introduce one concept.

Reply:
Oh, the core wording in [[basic.fundamental]](https://eel.is/c++draft/basic.fundamental) looks weird... It seems that currently `const int` is neither a signed integer type nor a standard integer type because the terms only refer to cv-unqualified types.

Reply:
Pre-requisite added here: https://github.com/llvm/llvm-project/pull/78086
I'll update this patch after the above is merged.